rust-runtime/aws-smithy-http-client/tests/proxy_tests.rs

```diff
@@ -0,0 +1,1105 @@
+
+    // Make request with disabled proxy (should go direct to our mock server)
+    let result = make_http_request_through_proxy(
+        proxy_config,
+        &format!("http://{}/get", direct_server.addr()),
+    )
+    .await;
+
+    let (status, body) = result.expect("Direct connection should succeed");
+    assert_eq!(status, StatusCode::OK);
+    assert_eq!(body, "direct connection");
+
+    // Verify the direct server received the request
+    let requests = direct_server.requests();
+    assert_eq!(
+        requests.len(),
+        1,
+        "Direct server should have received the request"
+    );
+    assert_eq!(requests[0].method, "GET");
+    // For direct connections, the URI might be just the path part
+    assert!(
+        requests[0].uri == format!("http://{}/get", direct_server.addr())
+            || requests[0].uri == "/get",
+        "URI should be either full URL or path, got: {}",
+        requests[0].uri
+    );
+}
+
+/// Tests HTTPS-only proxy configuration
+/// Verifies that HTTP requests bypass HTTPS-only proxies
+#[tokio::test]
+async fn test_https_proxy_configuration() {
+    let mock_proxy = MockProxyServer::with_response(StatusCode::OK, "https proxy response").await;
+
+    // Create a direct target server for HTTP requests
+    let direct_server =
+        MockProxyServer::with_response(StatusCode::OK, "direct http connection").await;
+
+    // Configure HTTPS-only proxy
+    let proxy_config = ProxyConfig::https(format!("http://{}", mock_proxy.addr())).unwrap();
+
+    // Test: HTTP request should NOT go through HTTPS-only proxy, should go direct
+    let target_url = format!("http://{}/api", direct_server.addr());
+    let result = make_http_request_through_proxy(proxy_config.clone(), &target_url).await;
+
+    // The HTTP request should succeed by going directly to our mock server
+    let (status, body) = result.expect("HTTP request should succeed via direct connection");
+    assert_eq!(status, StatusCode::OK);
+    assert_eq!(body, "direct http connection");
+
+    // Verify the HTTPS-only proxy received no requests
+    let proxy_requests = mock_proxy.requests();
+    assert_eq!(
+        proxy_requests.len(),
+        0,
+        "HTTP request should not go through HTTPS-only proxy"
+    );
+
+    // Verify the direct server received the request
+    let direct_requests = direct_server.requests();
+    assert_eq!(
+        direct_requests.len(),
+        1,
+        "Direct server should have received the HTTP request"
+    );
+}
+
+/// Tests all-traffic proxy configuration
+/// Verifies that both HTTP and HTTPS requests go through all-traffic proxies
+#[tokio::test]
+async fn test_all_traffic_proxy() {
+    let mock_proxy = MockProxyServer::with_response(StatusCode::OK, "all traffic proxy").await;
+
+    // Configure proxy for all traffic
+    let proxy_config = ProxyConfig::all(format!("http://{}", mock_proxy.addr())).unwrap();
+
+    // Test 1: HTTP request should go through the proxy
```

Comment:
Will there be more tests given this comment explicit says Test 1, like for HTTPS?